### PR TITLE
Accommodate 0735ecb33 in trace_dmu.h

### PR DESCRIPTION
**Pull request by akorn**
0735ecb33485e91a78357a274e47c2782858d8b9 renamed tx_waited to tx_dirty_delayed, but trace_dmu.h wasn't updated.

This is a minimal fix needed to enable the code to build, avoiding the error

include/sys/trace_dmu.h:65:31: error: ‘dmu_tx_t {aka struct dmu_tx}’ has
  no member named ‘tx_waited’
      __entry->tx_waited   = tx->tx_waited;
                               ^

Signed-off-by: András Korn <korn-github@elan.rulez.org>

### How Has This Been Tested?

It builds successfully on latest Debian sid, with the stock Debian kernel. It didn't build before.

I don't know how I could test the functionality of the impacted code.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

**Comment by akorn** [comment]
I only have a very basic understanding of the code and have no idea whether this is the proper fix.

File: include/sys/trace_dmu.h [hunk 1]
```diff
@@ -62,7 +62,7 @@ DECLARE_EVENT_CLASS(zfs_delay_mintime_class,
 	    __entry->tx_lastsnap_txg		= tx->tx_lastsnap_txg;
 	    __entry->tx_lasttried_txg		= tx->tx_lasttried_txg;
 	    __entry->tx_anyobj			= tx->tx_anyobj;
-	    __entry->tx_waited			= tx->tx_waited;
+	    __entry->tx_waited			= tx->tx_dirty_delayed;
```
Comment (dinatale2) [inline include/sys/trace_dmu.h]: The `printk` below looks like it's attempting to print out the fields of the `dmu_tx_t` structure. I think you should rename `tx_waited` to be `tx_dirty_delayed` in the in the `__entry` structure to be consistent.

Comment (behlendorf) [reply include/sys/trace_dmu.h]: Yes, please update all the instances of `tx_waited` in this macro to `tx_dirty_delayed`.

File: include/sys/trace_dmu.h [hunk 2]
```diff
@@ -73,7 +73,7 @@ DECLARE_EVENT_CLASS(zfs_delay_mintime_class,
 	    "anyobj %d waited %d start %llu wait_dirty %d err %i "
```
Comment (Fabian-Gruenbichler) [inline include/sys/trace_dmu.h]: `waited` should be replaced with `dirty_delayed` in the format string as well.

**Comment by Fabian-Gruenbichler** [comment]
@behlendorf (or whoever is responsible for the buildbot worker configs ;)) - it might make sense to configure one of the builders to have the appropriate kernel config/headers to compile with tracing support (see config/kernel-declare-event-class.m4), to prevent future changes that break it from slipping through like this one did?

**Comment by behlendorf** [comment]
@Fabian-Gruenbichler actually we did.  Unfortunately, we accidentally broke that builder in such a way that it was reporting success even though this was broken.  @dinatale2 has opened https://github.com/zfsonlinux/zfs-buildbot/pull/128 to resolve that infrastructure issue.  We'll get the bots updates once we sort out this issue and verify there aren't other issues which snuck in.  Sorry about that.

**Pull request closed without merging**